### ISSUE TITLE
fetch module path traversal

### ISSUE THREAD

**Issue by samdoran** [ansible/ansible#67793]
##### SUMMARY
CVE-2020-1735

Possibly related to CVE-2019-3828 (https://github.com/ansible/ansible/pull/52133/) 

The `fetch` module takes the `source` result from the `slurp` module, which came from the remote host. We don't really validate this path and it could have been manipulated by the remote host in a malicious way such that we end up a path similar to `/tmp/result_fetch/ansible1/../../../../../../../../../../../../../../../../home/<user>/.profile` for the source. This allows an attacker to place a file the contents of which they control.

**Relevant Code:**

https://github.com/ansible/ansible/blob/79dfae9624480f757795224732bfaa9ca1dd4543/lib/ansible/plugins/action/fetch.py#L83-L102

Suggested correction from the reporter:

- Don't use `%s/%s/%s` to compute the destination file or clean the last argument
- add the following check:

```python
target_dir = os.path.normpath(os.path.join(self._loader.path_dwim(dest), target_name))
dest = os.path.normpath(os.path.join(target_dir, source_local)
assert os.path.commonpath([target_dir, dest]) == target_dir
```

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

`lib/ansible/plugins/action/fetch.py`

##### ANSIBLE VERSION

```paste below
2.10
```

##### CONFIGURATION

```paste below
default
```

##### STEPS TO REPRODUCE

```yaml

```

##### ACTUAL RESULTS

```paste below

```

**Comment by samdoran** [comment]
`os.path.commonpath()` was added in Python 3.4 but is not available in Python 2.7, so we cannot use that. `os.path.commonprefix()` is available in Python 2.7 and should work, though.

**Comment by bcoca** [comment]
we can just remove those lines since we already do remote-expand previously, there is no need to use the return from slurp.
